plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -566,90 +517,84 @@ public String commit(String container, String repository, String tag, String com
+        final ExecConfig execConfig = new ExecConfig().withCmd(cmd);
+        if (!detach) {
+            execConfig.withAttachStderr(true).withAttachStdout(true);
+        }
+        final String entity = JsonHelper.toJson(execConfig, FIRST_LETTER_LOWERCASE);
+        headers.add(Pair.of("Content-Length", entity.getBytes().length));
+
+        try (DockerConnection connection = openConnection(dockerDaemonUri).method("POST")
+                                                                          .path("/containers/" + container + "/exec")
+                                                                          .headers(headers)
+                                                                          .entity(entity)) {
+            final DockerResponse response = connection.request();
             final int status = response.getStatus();
             if (status / 100 != 2) {
-                final String msg = CharStreams.toString(new InputStreamReader(response.getInputStream()));
-                throw new DockerException(String.format("Error response from docker API, status: %d, message: %s", status, msg), status);
+                throw new DockerException(getDockerExceptionMessage(response), status);
             }
-            String execId = JsonHelper.fromJson(response.getInputStream(), ExecCreated.class, null, FIRST_LETTER_LOWERCASE).getId();
-            return new Exec(cmd, execId);
+            return new Exec(cmd, parseResponseStreamAndClose(response.getInputStream(), ExecCreated.class).getId());
         } catch (JsonParseException e) {
-            throw new IOException(e.getMessage(), e);
-        } finally {
-            connection.close();
+            throw new IOException(e.getLocalizedMessage(), e);
         }
     }
 
-    public void startExec(String execId, LogMessageProcessor execOutputProcessor) throws IOException {
-        final DockerConnection connection = openConnection(dockerDaemonUri);
-        try {
-            final List<Pair<String, ?>> headers = new ArrayList<>(2);
-            headers.add(Pair.of("Content-Type", "application/json"));
-            final ExecStart execStart = new ExecStart().withDetach(execOutputProcessor == null);
-            final String entity = JsonHelper.toJson(execStart, FIRST_LETTER_LOWERCASE);
-            headers.add(Pair.of("Content-Length", entity.getBytes().length));
-            final DockerResponse response = connection.method("POST")
-                                                      .path(String.format("/exec/%s/start", execId))
-                                                      .headers(headers)
-                                                      .entity(entity)
-                                                      .request();
+    public void startExec(String execId, MessageProcessor<LogMessage> execOutputProcessor) throws IOException {
+        final List<Pair<String, ?>> headers = new ArrayList<>(2);
```

Comment:
I think in this case it is better to define <code>headers</code> variable above <code>580</code> line + move 
<code>headers.add(Pair.of("Content-Type", "application/json"));</code> after that declaration, then it will be like this:
```java
final List<Pair<String, ?>> headers = new ArrayList<>(2);
headers.add(Pair.of("Content-Type", "application/json"));
headers.add(Pair.of("Content-Length", entity.getBytes().length));
```